config/jsonconfig/config.go

```diff
@@ -0,0 +1,109 @@
+/*
+jsonconfig implements configuration, reading json into an ice Module.
+
+To use:
+
+1) Create the Options. List your configurable Interfaces. Each Interface
+can be backed by several named Implementations.
+2) Options.Parse parses bytes and creates a Configuration.
+  a) for each Interface, pick which Implementation.
+  b) json.Unmarshal the json into that Implementation
+  c) Implementation can now be used as a Module or json.Marshal'ed to print its configuration
+3) Configuration is an ice Module that installs all the configured Implementations
+*/
+package jsonconfig
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/scootdev/scoot/ice"
+)
+
+type Options map[string]Interface
+
+type Interface map[string]Implementation
+
+type Implementation interface {
+	// The Implementation needs to do 3 things:
+	// 1) parse the JSON config
+	// 2) add the relevant providers to the ice MagicBag
+	// 3) print its configuration
+	// 1 & 3 are handled implicitly (ugh) by json.(Un)marshal
+	// 2 is handled by being an ice Module
+	ice.Module
+}
+
+/*
+Example JSON of a Configuration:
+{
+ "Storage": {
```

Comment:
So my one concern with this config format is it couples the implementation type to the config for that type.
Basically assumes that all the implementations have the same config value.  We can address this later though